zcash_client_sqlite/src/wallet.rs

```diff
@@ -808,213 +808,440 @@ pub(crate) fn get_derived_account<P: consensus::Parameters>(
+
+                                    (scanned_notes * unscanned_range)
+                                        .checked_div(scanned_range)
+                                        .map(|extrapolated_unscanned_notes| {
+                                            last_scanned_tree_size + extrapolated_unscanned_notes
+                                        })
+                                })
+                                .flatten()
+                        });
+
+                    if let Some(tree_size) = tip_tree_size {
+                        Some(tree_size)
+                    } else if let Some(second_to_last_completed_subtree_height) =
+                        last_completed_subtree_index
+                            .checked_sub(1)
+                            .and_then(|subtree_index| {
+                                conn.query_row(
+                                    &format!(
+                                        "SELECT subtree_end_height
+                                        FROM {table_prefix}_tree_shards
+                                        WHERE shard_index = :shard_index"
+                                    ),
+                                    named_params! {":shard_index": subtree_index},
+                                    |row| {
+                                        Ok(row.get::<_, Option<_>>(0)?.map(BlockHeight::from_u32))
+                                    },
                                 )
-                            },
-                        ))
-                    },
-                )
-                .optional()?
-                .flatten())
-        }
+                                .transpose()
+                            })
+                            .transpose()?
+                    {
+                        let notes_in_complete_subtrees =
+                            (last_completed_subtree_index + 1) << shard_height;
+
+                        let subtree_notes = 1 << shard_height;
+                        let subtree_range = u64::from(
+                            last_completed_subtree_height - second_to_last_completed_subtree_height,
+                        );
+                        let unscanned_range =
+                            u64::from(chain_tip_height - last_completed_subtree_height);
+
+                        (subtree_notes * unscanned_range)
+                            .checked_div(subtree_range)
+                            .map(|extrapolated_incomplete_subtree_notes| {
+                                notes_in_complete_subtrees + extrapolated_incomplete_subtree_notes
+                            })
+                    } else {
+                        // There's only one completed subtree; its start height must
+                        // be the activation height for this shielded protocol.
+                        let scanned_notes = last_completed_subtree_index << shard_height;
```

Comment:
I think this also has the off-by-one error; if the completed subtree has index 0 (which is what you describe in the comment) then the value of `scanned_notes` will be zero.